### ISSUE TITLE
Incorrect serialization of fields with EnumType 

### ISSUE THREAD

**Issue by gregorwolf** [sapmentors/cap-community#73]
Hello CAP Team,

I try to create a mock service for the SAP Business One Service Layer (OData V4). In my sample project [sap-business-one-odata-cap branch cleared-b1-metadata](https://github.com/gregorwolf/sap-business-one-odata-cap/tree/cleared-b1-metadata) I've created an [init.js](https://github.com/gregorwolf/sap-business-one-odata-cap/blob/cleared-b1-metadata/srv/init.js) file that uses the sample data from the Sandbox Service hosted at  [api.sap.com](https://api.sap.com/api/Orders/resource?tag=Orders&path=get_Orders&method=get&opId=get_Orders). Unfortunately when calling that mock Service, the following error is returned:

> An error occurred during serialization of the entity. Serialization of the 'DocType' property failed. Invalid value dDocument_Items (JavaScript string). A number without decimals must be specified as value for type Edm.Int32.

I've investigated the issue and found that the Orders Entityset is based on the EntityType Document. There the DocType is defined as Type BoDocumentTypes:

```
      <EntityType Name="Document">
        <Key>
          <PropertyRef Name="DocEntry"/>
        </Key>
        <Property Name="DocEntry" Type="Edm.Int32"/>
        <Property Name="DocNum" Type="Edm.Int32"/>
        <Property Name="DocType" Type="SAPB1.BoDocumentTypes"/>
```

BoDocumentTypes itself is an EnumType:

```
      <EnumType IsFlags="false" Name="BoDocumentTypes" UnderlyingType="Edm.Int32">
        <Member Name="dDocument_Items" Value="0"/>
        <Member Name="dDocument_Service" Value="1"/>
      </EnumType>
```

When I replace now all Strings (Enum Member Name) in the sample data with the Enum Value (as I did with the Order 2) that can be requested from the mock service by calling http://localhost:4004/sapb1/Orders/2, there is no issue calling the service. 

But I think this isn't following the OData V4 standard. When I call the https://services.odata.org/TripPinRESTierService/(S(jgpg3ijmovlxq4w0qggz1rtg))/People then i.e.:

> "FavoriteFeature": "Feature1",

is returned. This element is based on the EnumType Feature:

```
<EnumType Name="Feature">
  <Member Name="Feature1" Value="0"/>
  <Member Name="Feature2" Value="1"/>
  <Member Name="Feature3" Value="2"/>
  <Member Name="Feature4" Value="3"/>
</EnumType>
```

But the value returned is the Name and not the Value.

Best regards
Gregor

**Comment by gregorwolf** [comment]
The OpenUI5 Issue [OData V4 Filtering with EnumType BoYesNoEnum fails #2390](https://github.com/SAP/openui5/issues/2390) is also using SAP Business One and is quite related to this.

**Comment by gregorwolf** [comment]
I've filed this also via the SAP Service Marketplace at 315576 / 2020.

**Comment by gregorwolf** [comment]
It is fixed with @sap/cds-dk version 1.8.4
